subsys/bluetooth/host/scan.c

```diff
@@ -694,8 +694,10 @@ void bt_hci_le_per_adv_sync_established(struct net_buf *buf)
 	}
 
 	if (!pending_per_adv_sync ||
-	    pending_per_adv_sync->sid != evt->sid ||
-	    bt_addr_le_cmp(&pending_per_adv_sync->addr, &evt->adv_addr)) {
+	    (!atomic_test_bit(pending_per_adv_sync->flags,
+			     BT_PER_ADV_SYNC_USE_LIST) &&
+	     ((pending_per_adv_sync->sid != evt->sid) ||
+	      bt_addr_le_cmp(&pending_per_adv_sync->addr, &evt->adv_addr)))) {
```

Comment:
The check here is correct, although I wish we could make it a bit more readable, but I don't have any suggestions. 

There's a little bit of indentation that should be fixed 
```suggestion
	    (!atomic_test_bit(pending_per_adv_sync->flags,
			      BT_PER_ADV_SYNC_USE_LIST) &&
	     ((pending_per_adv_sync->sid != evt->sid) ||
	       bt_addr_le_cmp(&pending_per_adv_sync->addr, &evt->adv_addr)))) {
```

Reply:
Only line 698 was missing a whitespace, fixed.